DancingGoat/Scripts/PreviewMode/PreviewMode.js

```diff
@@ -0,0 +1,107 @@
+﻿const previewModeToggleClassname = 'preview-mode-switch';
+const previewModeCookieName = 'IsPreviewMode';
+const previewApiKeyCookieName = 'PreviewApiKey';
+const enterApiKeyFirstMessage = 'Enter your Preview API key';
+const enterApiKeyNextMessage = 'Enter your Delivery Preview API key -- you can find it in Kentico Cloud under Project settings -> API keys';
+const enterApiKeyPromptTitle = 'Preview API key';
+
+const previewModeToggles = document.getElementsByClassName(previewModeToggleClassname);
+const isPreviewModeEnabled = getCookie(previewModeCookieName);
+for (toggle of previewModeToggles) {
+    toggle.checked = isPreviewModeEnabled === 'true' ? true : false;
+}
+
+function togglePreviewMode(isChecked, projectId, isFirstEnterOfApiKey) {
+    if (isChecked) {
+        const apiKey = getCookie(previewApiKeyCookieName);
+        if (apiKey === '') {
+            enterApiKey(isFirstEnterOfApiKey, projectId);
+        } else {
+            setPreviewModeEnabledCookie(isChecked);
+            location.reload();
+        }
+    } else {
+        setPreviewModeEnabledCookie(isChecked);
+        location.reload();
+    }
+}
+
+function enterApiKey(isFirstEnterOfApiKey, projectId) {
+    logAttemptToEnterPreviewKey(isFirstEnterOfApiKey);
+    const message = isFirstEnterOfApiKey ? enterApiKeyFirstMessage : enterApiKeyNextMessage;
+    const apiKey = prompt(message, enterApiKeyPromptTitle);
+    if (apiKey === null) {
+        const previewModeToggles = document.getElementsByClassName(previewModeToggleClassname);
+        for (toggle of previewModeToggles) {
+            toggle.checked = false
+        }
+    }
+
+    if (apiKey !== null) {
+        validateApiKey(apiKey, projectId);
+    }
+}
+
+function validateApiKey(apiKey, projectId) {
+    const deliverRequestUrl = 'https://preview-deliver.kenticocloud.com/' + projectId + '/items/home';
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', deliverRequestUrl, true);
+    xhr.setRequestHeader("authorization", `Bearer ${apiKey}`);
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === 4 && xhr.status === 200) {
+            logEnterPreviewKeyResult(true);
+            setPreviewModeEnabledCookie(true);
+            setPreviewApiKeyCookie(apiKey);
+            location.reload();
+        }
+
+        if (xhr.readyState === 4 && xhr.status === 401) {
+            logEnterPreviewKeyResult(false);
+            togglePreviewMode(true, projectId, false);
+        }
+    };
+    xhr.send();
+}
+
+
+function getCookie(cname) {
```

Comment:
Yes it's not ;). At first I wanted to use jquery's $.cookie() but it's not possible since cookie module must be added. So I decided to go this way.

ref.: https://www.w3schools.com/js/js_cookies.asp